packages/contracts-bedrock/test/setup/ForkLive.s.sol

```diff
@@ -51,20 +51,39 @@ contract ForkLive is Deployer {
 
     /// @notice Forks, upgrades and tests a production network.
     /// @dev This function sets up the system to test by:
-    ///      1. reading the superchain-registry to get the contract addresses we wish to test from that network.
-    ///      2. deploying the updated OPCM and implementations of the contracts.
-    ///      3. upgrading the system using the OPCM.upgrade() function.
+    ///      0. read the environment variable to determine the
+    ///         SUPERCHAIN_OPS_ALLOCS_PATH environment variable set from superchain ops.
+    ///      1. if the environment variable is set, load the state from the given path.
+    ///      2. read the superchain-registry to get the contract addresses we wish to test from that network.
+    ///      3. deploying the updated OPCM and implementations of the contracts.
+    ///      4. upgrading the system using the OPCM.upgrade() function if not using the applied state from superchain
+    /// ops.
     function run() public {
-        // Read the superchain registry and save the addresses to the Artifacts contract.
-        _readSuperchainRegistry();
-
-        // Now deploy the updated OPCM and implementations of the contracts
-        _deployNewImplementations();
+        string memory superchainOpsAllocsPath = vm.envOr("SUPERCHAIN_OPS_ALLOCS_PATH", string(""));
+
+        bool useOpsRepo = bytes(superchainOpsAllocsPath).length > 0;
+        if (useOpsRepo) {
+            console.log("ForkLive: loading state from %s", superchainOpsAllocsPath);
+            // Set the resultant state from the superchain ops repo upgrades
+            vm.loadAllocs(superchainOpsAllocsPath);
```

Comment:
how are these allocs generated?

Reply:
I think from `vm.dumpState`, but agreed we should have more details about that in the comments here

Reply:
https://github.com/ethereum-optimism/optimism/pull/14021/commits/802f3d489561d72389248442432ac4b04143c068